lib/workers/repository/model/semantic-commit-message.ts

```diff
@@ -0,0 +1,62 @@
+import { CommitMessage, CommitMessageJSON } from './commit-message';
+
+export interface SemanticCommitMessageJSON extends CommitMessageJSON {
+  scope?: string;
+  type?: string;
+}
+
+/**
+ * @see https://www.conventionalcommits.org/en/v1.0.0/#summary
+ *
+ * <type>[optional scope]: <description>
+ * [optional body]
+ * [optional footer]
+ */
+export class SemanticCommitMessage extends CommitMessage {
+  private static readonly REGEXP =
+    /(?<type>\w+)\s*[(]*(?<scope>\w+)*[)]*:\s*(?<description>[\w\s]*)/;
+
+  private scope?: string;
+  private type?: string;
+
+  static is(value: unknown): value is SemanticCommitMessage {
+    return value instanceof SemanticCommitMessage;
+  }
+
+  static fromString(value: string): SemanticCommitMessage {
+    const { groups } = value.match(SemanticCommitMessage.REGEXP);
+
+    const message = new SemanticCommitMessage();
+    message.setType(groups.type);
+    message.setScope(groups.scope);
+    message.setSubject(groups.description);
+
+    return message;
+  }
```

Comment:
Do we really need this? The regex is pretty bad!

Reply:
Yeah, we need it. I open for good suggestions about regex or another way to implement this function

Reply:
I have changed regex a bit, now it should be better